libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt

```diff
@@ -191,8 +205,7 @@ class MapboxNavigation(
         )
         tripSession = NavigationComponentProvider.createTripSession(
             tripService = tripService,
-            locationEngine = navigationOptions.locationEngine,
-            navigatorPredictionMillis = navigationOptions.navigatorPredictionMillis,
+            navigationOptions = navigationOptions,
```

Comment:
Do we want to pass in the whole `NavigationOptions` object around? I'd rather to pass only what's strictly necessary, even if that means adding one more parameter, I think is worth it, also `NavigationComponentProvider.createTripSession` is `internal`.

Reply:
Honestly wanted to make NavigationOptions available to all of Navigation